test/workers/test_async_toy.py

```diff
@@ -0,0 +1,254 @@
+import asyncio
+import pytest
+import threading
+
+from syft.workers.async_toy import ToyWorker
+from syft.workers.async_toy import AsynchronousWorker
+from syft.workers.async_toy import SynchronousWorker
+from syft.workers.async_toy import ThreadedSynchronousWorker
+from syft.workers.async_toy import ToyAction
+from syft.workers.async_toy import ToyMessage
+
+
+# The "set" action sets a boolean flag in the worker's object store. This is a stand-in for storing
+# a tensor and allows a similar style of checking for the availability of action dependencies.
+def test_set():
+    ToyWorker.reset_counter()
+
+    alice = ToyWorker(name="alice")
+    action = ToyAction("set", "test")
+
+    alice._execute_action(action)
+    assert alice.store["test"] is True
+
+
+# The "compute" action concatenates the worker's name with the current global counter of actions
+# executed and uses that value to set a flag in the worker's object store. This simulates performing
+# tensor operations and storing the output for later use.
+def test_compute():
+    ToyWorker.reset_counter()
+
+    alice = ToyWorker(name="alice")
+    action = ToyAction("compute", "test")
+
+    alice._execute_action(action)
+
+    assert alice.store == {"alice0": True}
+
+
+# The "send" action computes the same "name-counter" value as the "compute" action, but instead of
+# setting the corresponding flag in the local object store, it sends the value to a remote worker
+# and instructs the remote worker to set that flag in its object store. This simulates sending a
+# tensor value to a remote worker that may be used as an input to the remote worker's future
+# actions.
+def test_send():
+    ToyWorker.reset_counter()
+
+    alice = ToyWorker(name="alice")
+    bob = ToyWorker(name="bob")
+    ToyWorker.introduce(alice, bob)
+
+    action = ToyAction("send", "bob")
+
+    alice._execute_action(action)
+    assert bob.store["alice0"] is True
+
+
+# Unsupported actions should raise an exception
+def test_unknown_action():
+    ToyWorker.reset_counter()
+
+    alice = ToyWorker(name="alice")
+    action = ToyAction("flarble", "garb")
+
+    with pytest.raises(ValueError):
+        alice._execute_action(action)
+
+
+# The global action counter should be incremented every time any worker executes an action.
```

Comment:
I'm interested in this counter - what is its main function?

Reply:
It provides "sequential-ness" to actions by supplying a unique "id" for `set`/`compute` actions, which allows other actions to depend on their results.

Reply:
(In other words, it's a hack, but it works for a quick and dirty simulation.)